tools/mo/openvino/tools/mo/utils/ir_engine/compare_graphs.py

```diff
@@ -116,6 +116,27 @@ def compare_graphs(graph: Graph, graph_ref: Graph, last_node: str, last_node_ref
                                       ''.format(node.id, cur_node_type, attr))
                         continue
 
+                    def align_strided_slice_masks(curr_node: Node, rank: int):
+                        for mask_name in StridedSlice.get_mask_names():
+                            if type(curr_node[mask_name]) == int:
+                                curr_node[mask_name] = [curr_node[mask_name]]
+                            elif type(curr_node[mask_name]) == str:  # if mask is an empty string ''
+                                assert len(curr_node[mask_name]) == 0
+                                curr_node[mask_name] = []
+
+                            num_insertions = rank - len(curr_node[mask_name])
+                            curr_node[mask_name] = np.append(curr_node[mask_name], [0] * num_insertions).astype(int)
+
+                    # Need to align StridedSlice masks since such masks as [] and [0]; [] and [0,0]; [] and [0,0,0]
+                    # or [1] and [1,0]; [1] and [1,0,0] and so on for the input with rank 4 do exactly the same slicing and
+                    # should be treated as equal. Therefore, before attr comparison we align all masks to the input rank
+                    if cur_node_type == 'StridedSlice' and node.in_node(1).has('shape') \
+                            and node.in_node(1).shape is not None:
+                        from openvino.tools.mo.ops.strided_slice import StridedSlice
+                        input_rank = node.in_node(1).shape.item()
```

Comment:
Is it always valid? Can it be less than the real rank?

Reply:
BTW, here i called it `input_rank` but indeed is `slice_rank`. I fixed by renaming this variable to `slice_rank`
https://github.com/pavel-esir/openvino/blob/master/tools/mo/openvino/tools/mo/ops/strided_slice.py#L67C8-L68


If you mean by the real rank the `input_rank` (rank of the main tensor on port(0)), yes it can be less that it. But we should rely on `slice_rank` not to `input_rank`. 

if `begin=[]` then slice_rank will be 0. 
![image](https://github.com/openvinotoolkit/openvino/assets/5703039/1929c610-9814-44eb-a2aa-2c92616a8ec2)

I will add unit-tests and will ping when they are ready

Reply:
Unit-tests are ready. Indeed, `compare_graphs` is also called in almost every unit-test in python MO therefore we are sure that we didn't break anything